hazelcast/src/main/java/com/hazelcast/internal/monitor/impl/LocalExecutorStatsImpl.java

```diff
@@ -47,6 +48,8 @@
             .newUpdater(LocalExecutorStatsImpl.class, "totalStartLatency");
     private static final AtomicLongFieldUpdater<LocalExecutorStatsImpl> TOTAL_EXECUTION_TIME = AtomicLongFieldUpdater
             .newUpdater(LocalExecutorStatsImpl.class, "totalExecutionTime");
+
+    @Probe(name = EXECUTOR_METRIC_CREATION_TIME, unit = MS)
     private long creationTime;
```

Comment:
This field has to be marked as `volatile` or `final` in order to guarantee safe publication.

@blazember I can see that we use `volatile` for the same field in `LocalMapStatsImpl`, but not in `LocalQueueStatsImpl`, or `LocalTopicStatsImpl`, or `LocalPNCounterStatsImpl`, or many other stats classes. It looks like incorrect concurrent code to me. Maybe I'm missing something. Do you happen to know the reason for this?

Reply:
Good catch @puzpuzpuz :+1: These fields should be `volatile`. The reason is that I didn't catch it when I updated the `@Probe`'s. Don't know though why historically these are non-volatile, I think it is by mistake. According to the JLS (example 17.5-1), the reader threads could see 0 value for such fields even if initialized in the constructor - as you wrote. I raised #16778 to review and fix this.